high-level-caiman/src/lower/mod.rs

```diff
@@ -69,9 +67,8 @@ fn data_type_to_local_type(dt: &DataType) -> asm::TypeId {
 /// to the caiman assembly type id for the corresponding FFI type. For types
 /// that do not have FFI equivalents, this is the same as `data_type_to_local_type`.
 #[must_use]
-pub fn data_type_to_ffi_type(dt: &DataType) -> asm::TypeId {
-    use asm::TypeId;
-    data_type_to_ffi(dt).map_or_else(|| data_type_to_local_type(dt), TypeId::FFI)
+pub fn data_type_to_ffi_type(dt: &DataType) -> asm::FFIType {
+    data_type_to_ffi(dt).expect(&format!("Undefined type {:?}", dt))
```

Comment:
Sorry for the confusion!  I updated the documentation in `data_type_to_ffi_type`.  The behavior in `data_type_to_local_type` has not changed conceptually, however, because the change was on the AST side -- `Local` is no longer explicitly stated, and is just part of `TypedId`

Reply:
To clarify in a bit more detail -- I previously did not distinguish between the two at a parsing level, and sort of allowed you to use either anywhere.  This was causing a weird bug in the explicator, that as far as I can tell was just passing typechecking by sheer coincidence (the indices just always matched up in the conversion), _or_ some of the weird transformations being done "worked out" in a way I wasn't too interested in checking.  It's easier and less bug-prone to just differentiate in the AST as well.  This only changes the behavior of `data_type_to_ffi_type` and actually breaks nothing, since the only cases being removed were unreachable anyway.

Reply:
Sounds good, thanks. Although somehow I am not seeing the change on github?

Reply:
Wait nvm, ignore that lol